### ISSUE TITLE
Update DataBlock component usage to use the "Gathering data" prop

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4711]
## Feature Description

Once #4692 is implemented, we should convert the remaining `<DataBlock />` components to show the "Gathering data" UI. This involves updating the `<DataBlock />` components in `AdminBar*.js` and `WPDashboard*.js` files to use the new `gatheringData` prop. It may require updating the `<DataBlock />` component to deal with any differences in styling on the WP Dashboard and Admin Bar compared to the UI implemented in #4692.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The changes for this issue should only happen when the `zeroDataStates` feature flags are enabled. #4726
* All `<DataBlock />` instances in `AdminBar*.js` and `WPDashboard*.js` files should use the `gatheringData` prop after checking all relevant modules for `isGatheringData()`, similar to the approach used in #4692.
* The UI of the `<DataBlock />`, if needed, should be updated to work on the WP Dashboard and Admin Bar, as there could be differences in their UI compared to the Site Kit Dashboard. This should be noted in the IB.
* No changes made in #4692 should regress because of these changes.

## Implementation Brief
* Once #4692 has been merged.
* The changes below should only happen when the `zeroDataStates` feature flags are enabled; otherwise the component should behave as before.
* Update `assets/js/components/DataBlock.js`
	* to accept a new prop, `gatheringDataNoticeSize` which should have `normal` as default value. The value of `gatheringDataNoticeSize` should be passed as value to the `size` prop of the `GatheringDataNotice` component.
	* to add a new class name `googlesitekit-data-block--is-gathering-data` if the `gatheringData` prop is `true`.
* The following components should check if the module is gathering data via the `isGatheringData` selector and its value should be passed to the `gatheringData` prop of the `DataBlock` component along with `gatheringDataNoticeSize` set to `small` :
	* `AdminBarImpressions`
	* `AdminBarClicks`
	* `AdminBarUniqueVisitors`
	* `AdminBarSessions`
	* `WPDashboardUniqueVisitors`
	* `WPDashboardSessionDuration`
	* `WPDashboardImpressions`
	* `WPDashboardClicks`
* Ensure `assets/sass/components/global/_googlesitekit-gathering-data-notice.scss` is included in `assets/sass/wpdashboard.scss`
* There will be some additional styling tweaks needed to the `WPDashboard*` components where the blocks are not aligned properly if it's in the gathering data state. Hence, that's why a new class `googlesitekit-data-block--is-gathering-data` is added so that the necessary styles are added to prevent that.
* Stories for the above components should be updated to include examples when `isGatheringData` is `true`

### Test Coverage
* No new tests to be added.

## QA Brief

* Create a new site, and connect Analytics.
* Enable the `zeroDataStates` feature flag.
* Navigate to the WP Dashboard.
* Verify the _gathering data..._ text is displaying correctly:
![image](https://user-images.githubusercontent.com/18395600/156359492-e9c79d9c-9b6c-4c53-80ff-f3646a78a0eb.png)
* Navigate to a WP post and hover to show the Admin Bar.
* Verify the _gathering data..._ text is displaying correctly:
![image](https://user-images.githubusercontent.com/18395600/156359406-9223e052-2c69-41ac-8f2a-77db942c0ba7.png)

Storybook:
* Check the new stories added to Storybook are displaying correctly.
  * All **Gathering Data** stories under **AdminBarApp** are new.
  * All the stories under **WPDashboardClicks, WPDashboardImpressions, WPDashboardSessionDuration, and WPDashboardUniqueVisitors** are new.
* Storybook for `develop` branch: https://google.github.io/site-kit-wp/storybook/develop/
* ~~Storybook for this PR: https://google.github.io/site-kit-wp/storybook/pull/4903/
![image](https://user-images.githubusercontent.com/18395600/156362433-09b5760a-11df-4469-815a-7fe0ad0b6925.png)~~

## Changelog entry

* Update the WordPress Dashboard and Admin Bar with new "gathering data" UI for new sites.

**Comment by tofumatt** [comment]
Very minor, but how about `gatheringDataNoticeSize` or `gatheringDataNoticeStyle` instead of "format"?

The "format" term seems a bit technical and makes me think of some kind of "data format" when they're so close together. Heck, even `gatheringDataStyle` might be a bit more concise and usable. Are you cool with any of them as prop names?

Otherwise this sounds great 👍🏻

**Comment by asvinb** [comment]
@tofumatt All good to me 😁  I think `gatheringDataNoticeSize` makes more sense since it's more related to how the notice will be displayed. I've also tweaked the IB for #4692 to rename the `format` prop to the `size` prop.

**Comment by tofumatt** [comment]
Awesome, thanks!

IB ✅

**Comment by FlicHollis** [comment]
Moving this back to EB and un-assigning Eugene for now

**Comment by techanvil** [comment]
Hi @tofumatt, as I mentioned earlier there's a bit of a conflict between the AC/IB and the way some of these components work.

Looking at the AdminBar components that use Search Console data, i.e.  `AdminBarImpressions` and `AdminBarClicks`, it's evidently not possible for `isZeroData` to return `false` when `isGatheringData` is `true`.

https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/components/adminbar/AdminBarImpressions.js#L89-L90

That's because of this common condition in both functions. That plus the fact the same report is checked in both calls, i.e. the same report data is handed to `isZeroData` as is retrieved in `isGatheringData`.

_`isZeroData`:_
https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/modules/search-console/util/is-zero-report.js#L32-L34

_`isGatheringData`:_
https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/modules/search-console/datastore/report.js#L163-L165

The result of this, is that it's not possible for the `DataBlock` to be rendered with `gatheringData: true`, as the components will already have bailed out earlier rendering `WidgetReportZero`.

The situation is a bit different for those components that use Analytics data, i.e. `AdminBarSessions` and `AdminBarUniqueVisitors` - although, they also have the same condition, at a high level view, e.g.:

https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/components/adminbar/AdminBarSessions.js#L88-L89

However digging into it, there is a subtle difference in the conditions within `isZeroData` and `isGatheringData`. Plus, a key difference is the report data is **not** the same, the set of report args used for the report that's handed to `isZeroData` is different to those used to retrieve the report in `isGatheringData`. Which is something that seems like it possibly might not be correct, although maybe it just works in practice, I am not sure.

Comparable conditions for Analytics:

_`isZeroData`_:
https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/modules/analytics/util/is-zero-report.js#L32-L37

_`isGatheringData`_
https://github.com/google/site-kit-wp/blob/712dfc7d22cd69e446acf77c3bf995778ec09bc9/assets/js/modules/analytics/datastore/report.js#L318-L323

_Example Analytics report args passed to `isZeroData`:_
```js
{
    "startDate": "2020-12-31",
    "endDate": "2021-01-27",
    "compareStartDate": "2020-12-03",
    "compareEndDate": "2020-12-30",
    "dimensions": "ga:date",
    "limit": 10,
    "metrics": [
        {
            "expression": "ga:sessions",
            "alias": "Sessions"
        }
    ],
    "url": "https://www.sitekitbygoogle.com/blog/"
}

_Example Analytics report args retrieved in `isGatheringData`:
```js
{
    "dimensions": [
        "ga:date"
    ],
    "metrics": [
        {
            "expression": "ga:users"
        }
    ],
    "startDate": "2020-12-31",
    "endDate": "2021-01-27",
    "url": "https://www.sitekitbygoogle.com/blog/"
}
```

Additionally, the same `if ( isZeroReport( data ) && isGatheringData )` condition is present in all the `WPDashboard` components, however I have not dug into them in depth as yet.

The net of all the above is that we certainly need to address the situation for the Search Console related components, and we should probably apply the same to all of these components to be consistent and avoid possible problems with Analytics too.

Not sure what the best approach is, maybe we just avoid that condition when `unifiedDashboard` (or `zeroDataStates` ?, see question below) is enabled, i.e. `if ( ! featureEnabled && isZeroReport( data ) && isGatheringData )`, or whether we need to look at modifying `isZeroReport` and `isGatheringData` logic. Also not sure if the fact we're checking against different reports needs to be addressed.

---

One final, unrelated question: The IB mentions

> The changes below should only happen when the unifiedDashboard feature flag is enabled; otherwise the component should behave as before.

But the AC says:

> The changes for this issue should only happen when the zeroDataStates feature flag is enabled

Is this a typo in the IB?

---

Interested to hear your thoughts, thanks.

**Comment by tofumatt** [comment]
Re: the feature flags, it's not a typo but it's incomplete. Originally this work was meant to be done under the `unifiedDashboard` feature flag because it only applies to components used in the Unified Dashboard. But when we realised we couldn't complete it before Unified Dashboard rollout, we made a new flag. The point of mentioning both was to signify: "don't change any components that don't appear on the Unified Dashboard". But the changes in this issue should also be behind the new `zeroDataStates` feature flag.

So everything should, essentially, be behind both flags.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@techanvil this passes the QAB but a few observations:

1. I find that the font size is quite small, even more so on mobile. Admittedly, I am using a 13" MBP, but feel it could be a few pixels bigger. I found 13px was better than 11px. I realise for the WP Dashboard that makes it bigger than the title font size though. I also realise we don't have figma designs for this, so I included some screenshots. 

What are your thoughts? 

3. I checked Storybook but I noticed this 404s `Storybook for this PR: https://google.github.io/site-kit-wp/storybook/pull/4903/` Do you know what the correct URL should be? 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/156546409-f7df72b4-e9cc-46ac-95f4-29b23ca5888c.png)

![image](https://user-images.githubusercontent.com/73545194/156546439-84939678-6b7c-43c1-9d15-9e3e7e17003c.png)

![image](https://user-images.githubusercontent.com/73545194/156546466-162a3c1e-a2ec-44bc-a2b0-6467effa0ec7.png)

![image](https://user-images.githubusercontent.com/73545194/156546488-6c4309f5-276b-4f8b-9886-7b1369108c84.png)

![image](https://user-images.githubusercontent.com/73545194/156546508-d48b2b16-9d1d-43c0-a8f5-2bab1fdb7fa0.png)

</details>

**Comment by techanvil** [comment]
Hi @wpdarren, thanks for your comments here.

> 1. I find that the font size is quite small, even more so on mobile. Admittedly, I am using a 13" MBP, but feel it could be a few pixels bigger. I found 13px was better than 11px. I realise for the WP Dashboard that makes it bigger than the title font size though. I also realise we don't have figma designs for this, so I included some screenshots.
> 
> What are your thoughts?

This font size does seem a bit small. Unfortunately, I don't see a true representation of the scale from your screenshots, they are displaying quite large for me (probably because you screen grabbed on a retina display but I am viewing on a non-retina). Nonetheless, 11px is a small font size, and I'm trying to see where it's come from (I have simply copied the existing `small` notice size in this PR).

Tracing the font size back, the 11px notice size was introduced in [this PR](https://github.com/google/site-kit-wp/pull/4870/files#diff-646847fdb7898557ba8699c6cccd19ec3829002ceb19eea3cbb600be7b0c4adb). The relevant point in the IB states:

_* For the small size, refer to the styled of the notice displayed in the Acquisition section._

However it's not clear to me what notice in the Acquisition section is being referred to here. I've not seen anything with an 11px font size. I'd have guessed, _maybe_ this text, but it's 14px.

![image](https://user-images.githubusercontent.com/18395600/156576185-149802aa-48a0-408a-bed7-b3bab9a7277d.png)

Anyway, maybe all the above is slightly beside the point. If it's not a pleasant UX, then we should address it.

Given we don't have a design, and this small notice size is only being used here, by way of this issue, I think let's increase the font size, and then we can tweak if necessary once this has been reviewed by @marrrmarrr / @felixarntz.

I'll create a followup PR to bump the font size to 13px.

> 3. I checked Storybook but I noticed this 404s `Storybook for this PR: https://google.github.io/site-kit-wp/storybook/pull/4903/` Do you know what the correct URL should be?

Apologies, I should have realised that branch Storybook would be deleted once the PR is merged, and so is not relevant once it hits QA. I'll bear this in mind in future.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:** 

- On the WP dashboard, the `gathering data...` text is displaying correctly
- On the FE the `gathering data...` text is displaying correctly on the stats overview within top menu.
- The Storybook elements are displaying as expected.

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/156725046-c786d94a-6032-4b38-b066-77b976708b6d.png)

![image](https://user-images.githubusercontent.com/73545194/156725068-04db1d95-f58b-4395-8dc6-da3b99b7833f.png)

![image](https://user-images.githubusercontent.com/73545194/156725095-f6ba19a6-0fe5-4f0d-a21f-f99417cb0fc1.png)

</details>
